packages/resolve-query/src/index.js

```diff
@@ -3,32 +3,20 @@ import 'regenerator-runtime/runtime';
 function getExecutor({ eventStore, projection }) {
     const eventTypes = Object.keys(projection.eventHandlers);
     let state = projection.initialState || {};
-    let error = null;
 
-    const eventStream = eventStore.getStreamByEventTypes(eventTypes);
+    let result = null;
 
-    eventStream.on('readable', () => {
-        let event;
-        // eslint-disable-next-line no-cond-assign
-        while (null !== (event = eventStream.read())) {
-            const handler = projection.eventHandlers[event.type];
-            if (!handler) continue;
+    return async () => {
+        result =
+            result ||
+            eventStore.subscribeByEventType(eventTypes, (event) => {
+                const handler = projection.eventHandlers[event.type];
+                if (!handler) return;
 
-            try {
                 state = handler(state, event);
-            } catch (err) {
-                error = err;
-            }
-        }
-    });
-
-    eventStream.on('error', err => (error = err));
+            });
 
-    const storageReadyPromise = new Promise(resolve => eventStream.on('storageDone', resolve));
-
-    return async () => {
-        if (error) throw error;
-        await storageReadyPromise;
+        await result;
         return state;
```

Comment:
Unfortunately, no. User call stack with `query` function will not contain raised exception and ways to catch it, only by subscription to `uncatchedRejection` in global process variable, which is bad practice.
Exception will raise in some timeout/immediate stack on "thread", which reading events from database or event bus. Additionaly, depend on it implementation, all storage or bus provider can crash.